sql/catalyst/src/main/scala/org/apache/spark/sql/types/TimestampNTZType.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.types
+
+import scala.math.Ordering
+import scala.reflect.runtime.universe.typeTag
+
+import org.apache.spark.annotation.Unstable
+
+/**
+ * The timestamp without time zone type represents a local time in microsecond precision.
+ * It does not store or represent a time-zone.
+ * Its valid range is [0001-01-01T00:00:00.000000, 9999-12-31T23:59:59.999999].
+ *
+ * Please use the singleton `DataTypes.TimestampNTZType` to refer the type.
+ * @since 3.2.0
+ */
+@Unstable
+class TimestampNTZType private() extends AtomicType {
```

Comment:
Since it will be visible to users, let's discuss other names for the type:
- TimestampWithoutTZType - like in the standard
- Datetime - BigQuery: https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#datetime_type
- TimestampNoTzType
- LocalTimestampType

Could you elaborate why the name you selected is better.

Reply:
If we want to fully match the SQL standard, we should use `TimestmapWithoutTimeZoneType`.

Reply:
It's visible to non-SQL users, so the name does matter. But `TimestmapWithoutTimeZoneType` is really a bit too long to type. Maybe `TimestampNoTzType`?

Reply:
@MaxGekk I was to avoid the new type name being too long. 
`TimestampWithoutTZType` sounds great.